Update SetDesiredPropertyUpdateCallback to SetDesiredPropertyUpdateCa…

jierong opened the pull request:
…llbackAsync

Due to SetDesiredPropertyUpdateCallback is obsolete, we should use the updated version.

[Obsolete("Please use SetDesiredPropertyUpdateCallbackAsync.")]
        public Task SetDesiredPropertyUpdateCallback(DesiredPropertyUpdateCallback callback, object userContext);

azurecla commented:
Hi __@brentrossen__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

brentrossen commented:
Switched to private repo

jierong committed:
Update SetDesiredPropertyUpdateCallback to SetDesiredPropertyUpdateCallbackAsync

Due to SetDesiredPropertyUpdateCallback is obsolete, we should use the updated version.

[Obsolete("Please use SetDesiredPropertyUpdateCallbackAsync.")]
        public Task SetDesiredPropertyUpdateCallback(DesiredPropertyUpdateCallback callback, object userContext);
